### PR TITLE
GPC-173: Use bot PAT for auto fixing code

### PR DESCRIPTION
**Pull request by olevett**

### CODE REVIEW

File: .github/workflows/terraform-fmt-fix.yml [hunk 1]
```diff
@@ -16,7 +16,9 @@ jobs:
         working-directory: terraform
     steps:
       - name: Checkout code
-        uses: actions/checkout@v3
+        uses: actions/checkout@v3   
+        with:
+          token: ${{ secrets.PAT }}
```
Comment (ethanmills) [inline .github/workflows/terraform-fmt-fix.yml:21]: What does this give us? I'd assume it's the autocommit action that needed the token?

Comment (oskarwilliams378) [reply .github/workflows/terraform-fmt-fix.yml:21]: Apparently not, according to the docs for that action

**Review by ethanmills — APPROVED** [review]
Ok, that's just not how it works

File: .github/workflows/terraform-fmt-fix.yml (hunk 1 above)
Comment (oskarwilliams378) [reply .github/workflows/terraform-fmt-fix.yml:21]: https://github.com/stefanzweifel/git-auto-commit-action#commits-made-by-this-action-do-not-trigger-new-workflow-runs

Comment (olevett) [reply .github/workflows/terraform-fmt-fix.yml:21]: As far as I can tell, it means the checkout has the ability to push the code later, through whatever action. It's a little counterintuitive though...

**Pull request merged**